src/stacked_borrows.rs

```diff
@@ -326,31 +388,36 @@ impl<'tcx> Stack {
         global: &GlobalStateInner,
         alloc_history: &mut AllocHistory,
     ) -> InterpResult<'tcx> {
-        if global.tracked_pointer_tags.contains(&item.tag) {
+        if global.tracked_pointer_tags.contains(&item.tag()) {
             register_diagnostic(NonHaltingDiagnostic::PoppedPointerTag(
                 *item,
                 provoking_access.map(|(tag, _alloc_range, _size, access)| (tag, access)),
             ));
         }
 
-        if let Some(call) = item.protector {
-            if global.is_active(call) {
-                if let Some((tag, _alloc_range, _offset, _access)) = provoking_access {
-                    Err(err_sb_ub(
-                        format!(
-                            "not granting access to tag {:?} because incompatible item is protected: {:?}",
-                            tag, item
-                        ),
-                        None,
-                        tag.and_then(|tag| alloc_history.get_logs_relevant_to(tag, Some(item.tag))),
-                    ))?
-                } else {
-                    Err(err_sb_ub(
-                        format!("deallocating while item is protected: {:?}", item),
-                        None,
-                        None,
-                    ))?
-                }
+        if !item.protected() {
+            return Ok(());
+        }
+
+        if let Some(call_id) = global.protected_tags.get(&item.tag()) {
+            if let Some((tag, _alloc_range, _offset, _access)) = provoking_access {
+                Err(err_sb_ub(
+                    format!(
+                        "not granting access to tag {:?} because incompatible item is protected: {:?} (call {:?})",
```

Comment:
Yeah making this change a bit is totally fine.
```suggestion
                        "not granting access to tag {:?} because incompatible item {:?} is protected by call {:?}",
```
That seems better? The current thing looks a bit odd; I am open to other suggestions.
(Same for the `else` branch, of course.)

Reply:
The old messages are mostly born out of laziness. Don't assume I am wed to any of them. :)